src/Synthesiser.cpp

```diff
@@ -842,26 +842,134 @@ void Synthesiser::emitCode(std::ostream& out, const RamStatement& stmt) {
             }
 
             // create aggregation code
-            if (aggregate.getFunction() == RamAggregate::COUNT) {
+            if (aggregate.getFunction() == souffle::COUNT) {
```

Comment:
This if/else/switch block looks it would be simpler as a single switch statement, even with the mild duplication within some of the cases.
If it's important to keep in this form, then change line 854 to the empty string, and 862-865 to the default case (and if it's important to keep those specific empty cases, make the assert more informative.)